dali/operators/random/normal_distribution_op.cc

```diff
@@ -74,17 +74,18 @@ void NormalDistributionCpu::AssignTensorToOutput(workspace_t<CPUBackend> &ws) {
   auto &tp = ws.GetThreadPool();
   TYPE_SWITCH(dtype_, type2id, DType, NORM_TYPES, (
             for (int sample_id = 0; sample_id < batch_size_; ++sample_id) {
-              tp.DoWorkWithID(
-                  [&, sample_id](int thread_id) {
+              auto out_size = volume(output[sample_id].shape());
+              tp.AddWork(
+                  [&](int thread_id) {
                      distribution_t distribution(mean_[sample_id], stddev_[sample_id]);
                      auto ptr = output[sample_id].mutable_data<DType>();
-                     for (int64_t j = 0; j < volume(output[sample_id].shape()); j++) {
+                     for (int64_t j = 0; j < out_size; j++) {
```

Comment:
I wonder if you can capture `out_size` by the reference as this is a local tmp variable.